frontend/package.json

```diff
@@ -23,17 +22,20 @@
     "dayjs": "^1.11.13",
     "form-data": "4.0.0",
     "framer-motion": "10.16.16",
+    "node-fetch": "^3.3.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-error-boundary": "^4.0.13",
     "react-hook-form": "7.49.3",
     "react-icons": "5.0.1",
-    "ts-node": "^10.9.2"
+    "ts-node": "^10.9.2",
+    "util": "^0.12.5",
+    "whatwg-fetch": "^3.6.20"
```

Comment:
We shouldn't have node-fetch, ts-node, util, or whatwg-fetch in this repository. I'm not sure why they're here. `node-fetch` is something you'd add to a node.js repo. We don't need `ts-node` on the frontend either. This is likely related to the jest config environment issues.

Reply:
Unfortunately, Jest requires ts-node (in devDependencies, of course) to read TypeScript config files (if we are aiming for a TypeScript config in our project). Here's the official documentation: https://jestjs.io/docs/configuration

"To read TypeScript configuration files, Jest requires [ts-node](https://npmjs.com/package/ts-node). Make sure it is installed in your project."

We probably should have chosen Vitest.

Reply:
The other mentioned libs by Jake can be dropped.

Reply:
Nit, I don't love the idea of introducing a dependency just to have a configuration file written in TS. In the past I've done this using json in jest.config.json.

Reply:
https://jestjs.io/docs/configuration

Reply:
To read TypeScript configuration files Jest requires [ts-node](https://npmjs.com/package/ts-node). Make sure it is installed in your project.

Reply:
@DmitriyMalayev :  
These libraries should be removed from the `package.json`
*  node-fetch
* util
*  whatwg-fetch 
These seem to be the kinds of repos you would need to build out a server application.  This is only the frontend.  Please 
1. remove these from your `package.json`
2. rm -rf `node_modules`
3.  `npm install`
4. `npm test` or `npx jest`
Everything still works

   
(5) The following is a devDependency.  That means only the code developers are using it.  ts* files transform TypeScript to JavaScript for developers.  But on the web application, their browsers only run JS not TS.  So it's not a browser dependency
, ts-node, 
1.  remove ts-node from your `package.json`
2.  `npm install ts-node --save-dev`  (or `-D`)
3. Check your package json under `devDependencies` to make sure it's there
4. `rm -rf node_modules`
5. npm install
6. npm test

Reply:
addressed